### ISSUE TITLE
Integration with drag and drop API? 

### ISSUE THREAD

**Issue by IanBellomy** [WICG/view-transitions#234]
It'd be cherry to be able to include a drag-and-drop preview image in a view transition for a concise 'animate the dragged content to its final resting location'.

**Comment by bramus** [comment]
You can make a View Transition respond to drag by pausing all animations and then controlling them by monitoring the dragDelta. See https://front-end.social/@bramus/111800601352108818 for an exploration of this.

**Comment by IanBellomy** [comment]
Apologies for the confusion, let me clarify with an example of an acute limitation.

Here is an example of visual jank resulting from a 'naive' use of view transitions with the drag and drop API. 

https://github.com/user-attachments/assets/510d54c4-5958-40a5-8b8f-2fdc473e91d1

Attend to how, when the user makes a selection by releasing the content, the dragged content disappears, and the origin element of the drag then animates. 

For interfaces like that depicted, the ideal is for the drag preview image to animate to the resting location but the view transition API has no access to drag preview images. If it did, this could be very simple to implement. But in that absence a bespoke solution for drag previews is needed.  

Hope this clarifies. ✌️

**Comment by bramus** [comment]
Thanks for the video. Very clear now :)

**Comment by IanBellomy** [comment]
Bluesky User vtbag.dev was kind enough to create an example which shows the desired behavior while also exposing  currently inescapable jank.

Codepen: https://codepen.io/martrapp/pen/dPywMNR

Note the element visibility blink on drop -- there's no way around this currently without a bespoke JS drag and drop implementation.

https://github.com/user-attachments/assets/f3162555-6df6-423c-a738-9e30214064cf

**Comment by bramus** [comment]
Thanks for sharing the demo.

@vmpstr This issue may be closed as question answered.

**Comment by vmpstr** [comment]
Unsure if the question was answered or not. The glitch in question here seems to reproduce both of Chrome, Firefox, and Safari. This makes me believe that this is some inherent issue in timing that should be discussed in a spec venue, although I don't know how a drag-and-drop timing relates to view transition timing.

@IanBellomy the best course of action, imho, is to open an issue at https://github.com/w3c/csswg-drafts/issues pointing to [css-view-transitions-1] or 2.

This repo should be ultimately archived in favor of raising issues with the spec at the CSSWG

**Comment by IanBellomy** [comment]
> This issue may be closed as question answered.

Er, there is no question here; This is a feature request basically.  

> I don't know how a drag-and-drop timing relates to view transition timing.

Maybe you could help me here as the issue is cross domain -- the VT API doesn't touch drag previews and DND API doesn't touch transitions and so UI _idioms_ around 'drag and drop preview transitions' fall in a gap such that a common idiom is not possible to produce without jank.

Should I just post in csswg-drafts/issue and point to both APIs or is there a way to tag cross-topic issues maybe?

**Comment by vmpstr** [comment]
Yeah, you can absolutely reference multiple specs in the issues. I think this is primarily about view-transitions not working with existing drag-and-drop so that's why I suggested a view transition category. 

Admittedly I haven't looked in detail at the demo, but the idea of view transition is it takes a current dom as a captured state which is used for animation. My understanding of drag and drop is that the drag image is not actually in the dom. One likely outcome of that is that when we trigger a start view transition, the frame produced that constitutes the capture does not include the drag image. Presumably that's captured separately and then positioned via ::view-transition-group styling to the spot where the old image used to be? (that's where I have to dig into the demo).

I'm not sure what's involved in supporting this correctly, but CSSWG has experts that should weigh in

**Comment by IanBellomy** [comment]
Yea, your description is my understanding -- AFAIK the drag preview is ultimately handled by the OS, even if something image-able can be provided for that, and VTs don't have any access to it as is.

**Comment by IanBellomy** [comment]
> is to open an issue at https://github.com/w3c/csswg-drafts/issues pointing to [css-view-transitions-1] or 2.

🤦 Apparently I did just that two years ago:

https://github.com/w3c/csswg-drafts/issues/11221
